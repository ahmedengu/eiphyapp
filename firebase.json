{
  "database": {
    "rules": "database.rules.json"
  },
  "firestore": {
    "rules": "firestore.rules",
    "indexes": "firestore.indexes.json"
  },
  "functions": {
    "source": "functions"
  },
  "hosting": {
    "public": "out",
    "rewrites": [
      {
        "source": "**/**",
        "function": "next"
      }
    ],
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ]
  },
  "storage": {
    "rules": "storage.rules"
  },
  "emulators": {
    "functions": {
      "port": 5001
    },
    "firestore": {
      "port": 8080
    },
    "database": {
      "port": 9000
    },
    "hosting": {
      "port": 5000
    },
    "pubsub": {
      "port": 8085
    }
  },
  "firestore": {
    "rules": "firestore.rules",
    "indexes": "firestore.indexes.json"
  }
}
