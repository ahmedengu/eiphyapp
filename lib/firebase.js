import * as firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/storage';
import 'firebase/firestore';
import 'firebase/analytics';
import 'firebase/database';

const config = {
  apiKey: 'AIzaSyBhQc1hhUM_WfrOm1uchP18z8cayNuawFo',
  authDomain: 'eiphyappfinal.firebaseapp.com',
  databaseURL: 'https://eiphyappfinal.firebaseio.com',
  projectId: 'eiphyappfinal',
  storageBucket: 'eiphyappfinal.appspot.com',
  messagingSenderId: '851352675581',
  appId: '1:851352675581:web:2c102612e179aba8ee8a94',
  measurementId: 'G-6EKXZDW9MS',
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
  process.browser && firebase.analytics();
}

export default firebase;
export const db = firebase.firestore();
export const storage = firebase.storage();
