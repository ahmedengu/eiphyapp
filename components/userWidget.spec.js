import { mount } from 'enzyme';
import UserWidget from './userWidget';
import firebase from '../lib/firebase';

jest.mock('../lib/firebase', () => ({ auth: jest.fn(() => ({ onAuthStateChanged: jest.fn() })) }));

function setup(props = {}) {
  jest.clearAllMocks();
  return mount(<UserWidget {...props} />);
}

describe('UserWidget Tests', () => {
  it('should work', () => {
    const actual = setup();
    expect(actual).toBeTruthy();
  });

  it('should match snapshot', () => {
    const actual = setup();
    expect(actual.html()).toMatchSnapshot();
  });

  it('should call auth', () => {
    setup();
    expect(firebase.auth).toBeCalledTimes(1);
  });
});
