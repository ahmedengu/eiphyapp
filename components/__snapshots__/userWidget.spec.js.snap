// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserWidget Tests should match snapshot 1`] = `
<button style="display: inline-block; border: 0px; border-radius: 3px; box-shadow: rgba(0, 0, 0, 0.5) 0 1px 2px; color: rgb(255, 255, 255); cursor: pointer; font-size: 19px; margin: 5px; overflow: hidden; padding: 0px 10px; user-select: none; height: 50px; background: rgb(59, 89, 152); background-color: rgb(62, 62, 62); text-align: center;"
        class="login"
>
  <div style="align-items: center; display: flex; height: 100%;">
    <div style="display: flex; justify-content: center; min-width: 26px;">
      <svg aria-hidden="true"
           focusable="false"
           data-prefix="fas"
           data-icon="user-friends"
           class="svg-inline--fa fa-user-friends fa-w-20 fa-1x fa-swap-opacity "
           role="img"
           xmlns="http://www.w3.org/2000/svg"
           viewbox="0 0 640 512"
           color="white"
      >
        <path fill="currentColor"
              d="M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z"
        >
        </path>
      </svg>
    </div>
    <div style="width: 10px;">
    </div>
    <div style="text-align: left; width: 100%;">
      Login
    </div>
  </div>
</button>
`;
